src/controllers/zodiac.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import {
+  GetLastElementJSNatFromSignData,
+  getTodayAllSignRecordsFromDB,
+} from "../dal/zodiacdeep";
+import { ISignRecord } from "../interfaces/signs";
+import { getText } from "../helpers/arrayProcessing";
+
+export async function getZodiac(req: Request, res: Response): Promise<void> {
+  {
+    const sign = req?.query?.sign;
+    let totalResult = "";
+
+    if (typeof sign !== "undefined") {
+      if (Signs_ZodiacDirBG.includes(sign as string)) {
```

Comment:
This can be combined with the upper if statement, so it becomes:

if (typeof sign !== 'undefined' && Signs_ZodiacDirBG.includes(sign))